src/LineageTree/tree_styles.py

```diff
@@ -189,74 +189,63 @@ def get_norm(self):
-                )
-                if max_number_fragments > 0:
-                    current_node_lengths = self.node_length[
-                        :max_number_fragments
-                    ].copy()
-                    length_middle_node = (
-                        len(cycle) - sum(current_node_lengths) * 2
-                    )
-                    times_tmp = (
-                        current_node_lengths
-                        + [length_middle_node]
-                        + current_node_lengths[::-1]
-                    )
-                    pos_all_nodes = np.concatenate(
-                        [[0], np.cumsum(times_tmp[:-1])]
-                    )
-                    track = cycle[pos_all_nodes]
-                    self.out_dict.update(
-                        {k: [v] for k, v in zip(track[:-1], track[1:])}
-                    )
-                    self.times.update(zip(track, times_tmp))
-                else:
-                    for i, cell in enumerate(cycle[:-1]):
-                        self.out_dict[cell] = [cycle[i + 1]]
-                        self.times[cell] = 1
-                current = cycle[-1]
-                _next = self.lT[current]
-                self.times[current] = 1
-                if _next and self.lT.time[_next[0]] <= self.end_time:
-                    to_do.extend(_next)
-                    self.out_dict[current] = _next
-                else:
-                    self.out_dict[current] = []
-
+            if _next == [current]:
+                _next = None
+            if _next and self.lT.time[_next[0]] <= self.end_time:
+                self.out_dict[current] = _next
+                to_do.extend(_next)
+            else:
+                self.out_dict[current] = []
+            self.times[current] = self.downsample
         return self.out_dict, self.times
 
     def get_norm(self):
-        return len(self.lT.get_sub_tree(self.root, end_time=self.end_time))
+        return sum(self.times.values())
 
     def delta(self, x, y, corres1, corres2, times1, times2):
-        return super().delta(x, y, corres1, corres2, times1, times2)
+        if x is None and y is None:
+            return 0
+        if x is None:
+            return self.downsample
+        if y is None:
+            return self.downsample
+        return 0
+
+
+class normalized_tree(simple_tree):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def delta(self, x, y, corres1, corres2, times1, times2):
+        if x is None and y is None:
+            return 0
+        if x is None:
+            return 1
+        if y is None:
+            return 1
+        return abs(times1[corres1[x]] - times2[corres2[y]]) / (
+            times1[corres1[x]] + times2[corres2[y]]
+        )
+
+    def get_norm(self):
+        return len(
+            self.lT.get_all_branches_of_node(self.root, end_time=self.end_time)
+        )
 
 
 class full_tree(abstract_trees):
```

Comment:
Is the "full" tree necessary? Could it be downsampled at 1?

Reply:
I agree it could be downsample= 1, but maybe it is easier for users to understand it this way. I don't have any strong opinions on that, but if you want, I will make it downsample 1.